build-coordinator/src/main/java/org/jboss/pnc/core/builder/BuildScheduler.java

```diff
@@ -29,5 +29,6 @@
  * @author <a href="mailto:matejonnet@gmail.com">Matej Lazar</a>
  */
 public interface BuildScheduler {
+    boolean hasBeenBuilt(BuildTask buildTask);
```

Comment:
Build coordinator should be the one who decides if we need to schedule a build or not. BuildScheduler is an abstraction which allows to select the executor. Executor doesn't hold the information of what was build and what not, its responsibility is to run the build and other tasks around it, trying to be as stateless as possible.